lib/elixir/test/elixir/calendar/date/range_test.exs

```diff
@@ -0,0 +1,46 @@
+Code.require_file "../../test_helper.exs", __DIR__
+
+defmodule Date.RangeTest do
+  use ExUnit.Case, async: true
+  doctest Date.Range
+
+  setup do
+    { :ok, range: Date.range(~D[2000-01-01], ~D[2001-01-01]) }
+  end
+
+  describe "Enum.member?/2" do
+    test "for ascending range", %{range: range} do
+      assert Enum.member?(range, ~D[2000-02-22])
+      refute Enum.member?(range, ~D[2002-01-01])
+    end
+
+    test "for descending range", %{range: range} do
+      assert Enum.member?(range, ~D[2000-02-22])
+      refute Enum.member?(range, ~D[1999-01-01])
+    end
+  end
+
+  describe "Enum.count/1" do
+    test "counts days in range", %{range: range} do
+      assert Enum.count(range) == 367
+    end
+  end
+
+  describe "Enum.reduce/3" do
+    test "acts as a normal reduce" do
+      range = Date.range(~D[2000-01-01], ~D[2000-01-03])
+      fun = fn (date, acc) -> acc ++ [date] end
+
+      assert Enum.reduce(range, [], fun) == [~D[2000-01-01], ~D[2000-01-02], ~D[2000-01-03]]
+    end
+  end
+
+  test "both dates must have matching calendars" do
+    first = ~D[2000-01-01]
+    last = NaiveDateTime.new(2012, 1, 1, 1, 1, 1)
```

Comment:
this test states it fails for different calendars, but it fails because `NaiveDateTime` is not a `Date`. There's a `Calendar.Julian` that is used for tests - perhaps it would be more appropriate to use that, and have a different error message when someone passes non-dates?